bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentProvider.java

```diff
@@ -15,34 +15,41 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.Map;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.core.common.registry.AbstractProvider;
 import org.openhab.core.common.registry.ManagedProvider;
+import org.openhab.core.config.core.ConfigParser;
 import org.openhab.core.storage.Storage;
 import org.openhab.core.storage.StorageService;
 import org.openhab.core.ui.components.RootUIComponent;
+import org.openhab.core.ui.components.UIProvider;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
 /**
  * A namespace-specific {@link ManagedProvider} for UI components.
  *
  * @author Yannick Schaus - Initial contribution
+ * @author Jan N. Klug - Refactored to component factory
  */
 @NonNullByDefault
+@Component(factory = "org.openhab.core.ui.component.provider.factory")
 public class UIComponentProvider extends AbstractProvider<RootUIComponent>
         implements ManagedProvider<RootUIComponent, String>, UIProvider {
```

Comment:
Probably late to the "party" but I think it would be more logical to call this `UIProvider` interface `UIComponentProvider` (as it provides UICompnents, not UIs) and this `UIComponentProvider` class something like `UIComponentProviderImpl`.

Reply:
Good point. Since this was internal only, we can change it without issues.

Reply:
> this UIComponentProvider class something like UIComponentProviderImpl.

Or maybe even `ManagedUIComponentProvider[Impl]` since the pattern seems to be `ManagedThingProvider implements ThingProvider`, `ManagedItemProvider implements ItemProvider` etc.